cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/RunRecordMonitorService.java

```diff
@@ -115,20 +120,12 @@ public Count addRequestAndGetCount(ProgramRunId programRunId) throws Exception {
       launchingCount = launchingQueue.size();
     }
 
-    List<ProgramRuntimeService.RuntimeInfo> list = runtimeService
-      .listAll(ProgramType.WORKFLOW,
-               ProgramType.SERVICE,
-               ProgramType.MAPREDUCE,
-               ProgramType.SPARK,
-               ProgramType.WORKER,
-               ProgramType.CUSTOM_ACTION);
-
-    int runsCount = (int)
-      list.stream().filter(r -> isRunning(r.getController().getState().getRunStatus())).count()
-      + launchingCount;
-
-    LOG.info("Counter has {} concurrent launching and {} concurrent runs.", launchingCount, runsCount);
-    return new Count(launchingCount, runsCount);
+    int runningCount = getWorkflowsRunningCount();
+
+    emitMetrics(Constants.Metrics.FlowControl.WORKFLOWS_LAUNCHING, launchingCount);
```

Comment:
This metric is already added in addRequest() . Did you mean to add WORKFLOWS_RUNNING?

Reply:
Yes, it does not look like additional emit is needed at all.

Reply:
Thanks for the catch. Done!